qiskit_nature/second_q/circuit/library/ansatzes/succd.py

```diff
@@ -106,27 +107,42 @@ def include_singles(self, include_singles: Tuple[bool, bool]) -> None:
         """Sets whether to include single excitations."""
         self._include_singles = include_singles
 
+    @property
+    def mirror(self) -> bool:
+        """Whether use SUCC_full"""
+        return self._mirror
+
+    @mirror.setter
+    def mirror(self, mirror: bool) -> None:
+        """Sets whether to use SUCC_full."""
+        self._mirror = mirror
+
+    def _check_length_excitations_list(self, operators, excitation_list):
+        self._excitation_list = self.generate_excitations(
+            self.num_spin_orbitals,
+            self.num_particles,
+        )
+        self.operators = operators
```

Comment:
You're not actually performing any of the filtering here. You should be doing the equivalent operation to the one implemented in `UCC`. However, since this class now behaves differently as part of `_build_fermionic_excitation_ops`, so should this method.

To be more concrete:
- there is no need to call `self.generate_excitations` here. You already get that list as an input
- that said, you actually need to use the `exctiations_dictionary` which you are building out during `_build_fermionic_excitation_ops` (that means you need to store it as `self._excitations_dict` in order to be able to access it here)
- you then need to do a similar `zip(...)` operation as in the `UCC` class, but using the dictionary values instead

If this is not clear, please let me know and I will try to provide a pseudocode for you